### PR TITLE
Create cambridge.json

### PR DESCRIPTION
**Pull request by Sam-Spencer**
Not entirely sure if I've done this correctly, but I'd like to add Cambridge publications to the service...

### CODE REVIEW

File: src/cambridge.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "publisher": "Cambridge Core",
+  "publisher_parent": "Cambridge University Press",
```
Comment (sckott) [inline src/cambridge.json]: change to `"publisher_parent": null`

a parent is when there's a parent publisher

File: src/cambridge.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "publisher": "Cambridge Core",
```
Comment (sckott) [inline src/cambridge.json]: change to `"publisher": "cambridge"`

the values in `publisher` are keys we use to negotiate user  inputs and construct responses

File: src/cambridge.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "components": {
+    "html": null,
+    "doi": {
+      "regex": ".+"
+    }
+  },
+  "use_crossref_links": false,
+  "cookies": true,
+  "regex": null,
+  "open_access": true,
```
Comment (sckott) [inline src/cambridge.json]: cambridge isn't generally OA even if they have some OA articles, so i'd change to `"open_access": false`

File: src/cambridge.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  "publisher_member": null,
+  "crossref_member": null,
+  "prefixes": ["10.1017"],
+  "urls": {
+    "pdf": "https://www.cambridge.org/core/services/aop-cambridge-core/content/view/%s.pdf"
+  },
+  "components": {
+    "html": null,
+    "doi": {
+      "regex": ".+"
```
Comment (sckott) [inline src/cambridge.json]: `.+` doesn't quite work. this publisher wants just the part after the prefix and slash. so this should work: `[A-Za-z0-9\\-\\.]+$` though I haven't tested it very widely, you might want to do that

File: src/cambridge.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "publisher": "Cambridge Core",
+  "publisher_parent": "Cambridge University Press",
+  "publisher_member": null,
+  "crossref_member": null,
+  "prefixes": ["10.1017"],
+  "urls": {
+    "pdf": "https://www.cambridge.org/core/services/aop-cambridge-core/content/view/%s.pdf"
```
Comment (sckott) [inline src/cambridge.json]: Can you give some examples that work?  This URL doesn't work at least for the few DOIs I tried., e.g, this https://www.cambridge.org/core/services/aop-cambridge-core/content/view/10.1017/s0031819114000357.pdf doesn't work . This URL https://www.cambridge.org/core/services/aop-cambridge-core/content/view/S0031819114000357 does work for that DOI, but it's paywalled so if you don't have access it won't work 

for this publisher seems like the pattern should work , BUT they do seem to be a mess, e.g., 

this DOI 10.1557/opl.2014.876 has a link that looks like https://www.cambridge.org/core/services/aop-cambridge-core/content/view/S1946427414008768 - so completely unrelated to its DOI for the most part at least

File: src/cambridge.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+  "publisher": "Cambridge Core",
+  "publisher_parent": "Cambridge University Press",
+  "publisher_member": null,
+  "crossref_member": null,
+  "prefixes": ["10.1017"],
```
Comment (sckott) [inline src/cambridge.json]: please include all prefixes , can find them at http://api.crossref.org/members/56/

```
"prefixes": [
    "10.7313",
    "10.7135",
    "10.1557",
    "10.3815",
    "10.1375",
    "10.5949",
    "10.5948",
    "10.1017",
    "10.3189",
    "10.4039"
  ]
```

**Review by sckott — CHANGES_REQUESTED** [review]
thanks much @Sam-Spencer !!! - some requests for change

**Comment by sckott** [comment]
any thoughts @Sam-Spencer ?

File: src/cambridge.json (hunk 3 above)
Comment (Sam-Spencer) [reply src/cambridge.json]: Hm, it seems you are correct... I proposed this commit while on university internet (which seems to unlock most publishers, making it difficult to tell who actually has OA).

File: src/cambridge.json (hunk 5 above)
Comment (Sam-Spencer) [reply src/cambridge.json]: I fixed the other issues you noted, but will have to address this in a bit...

File: src/cambridge.json (hunk 3 above)
Comment (sckott) [reply src/cambridge.json]: makes sense

File: src/cambridge.json (hunk 5 above)
Comment (sckott) [reply src/cambridge.json]: okay

**Comment by sckott** [comment]
@Sam-Spencer just ping me when you're all done

**Pull request closed without merging**